source/common/upstream/upstream_impl.h

```diff
@@ -800,9 +800,11 @@ class ClusterInfoImpl : public ClusterInfo,
     }
     return *lb_original_dst_config_;
   }
-  const absl::optional<envoy::config::core::v3::TypedExtensionConfig>&
-  upstreamConfig() const override {
-    return upstream_config_;
+  OptRef<envoy::config::core::v3::TypedExtensionConfig> upstreamConfig() const override {
```

Comment:
nit: I feel like we should be returning an `OptRef<const envoy::config::core::v3::TypedExtensionConfig>` given that this is a const method.

Would that work at call-sites?

The version checked in now actually returned a const optional ref, so I think this PR should also return a config that cannot be modified by the caller.

Reply:
+1